MCLOUD-7072: Add CLI command for generating .magento.env.yaml (#8040)
magento/devdocs: src/cloud/project/magento-env-yaml.md

```

Make any corrections, commit, and push the changes. If you do not receive an error message, then the changes to your configuration file pass the validation.

## Create configuration file from CLI

> Sometimes it requires to generate the configuration file `.magento.env.yaml` from CLI, for this purpose `ece-tools` contains two CLI commands.

Reviewer: ```suggestion
Sometimes it is required to generate the configuration file `.magento.env.yaml` from CLI, for this purpose `ece-tools` contains two CLI commands.
```

> Sometimes it is required to generate the configuration file `.magento.env.yaml` from CLI, for this purpose `ece-tools` contains two CLI commands.

Reviewer 2: ```suggestion
You can use the following `ece-tools` CLI commands to generate a `.magento.env.yaml` configuration file for a Cloud environment.
```

```bash
php ./vendor/bin/ece-tools cloud:config:create     # Creates a new configuration file
php ./vendor/bin/ece-tools cloud:config:update     # Updates values in the configuration file
```

> Both commands accepts one argument - configuration in JSON format.

Reviewer: ```suggestion
Both commands accept one argument - configured in JSON format.
```

> For example, the next command

Reviewer: ```suggestion
For example, the next command :
```

```bash
php vendor/bin/ece-tools cloud:config:create '{"stage":{"build":{"SCD_THREADS":5}, "deploy":{"CLEAN_STATIC_FILES":false}}}'
```

> will create the next configuration file `.magento.env.yaml`

Reviewer: ```suggestion
will create a new configuration file `.magento.env.yaml`
```

```yaml
stage:
  build:
    SCD_THREADS: 5
  deploy:
    CLEAN_STATIC_FILES: false
```

> If after that you run `cloud:config:update` command 

Reviewer: ### 
```suggestion
If after that you run `cloud:config:update` command :
```

```bash
php vendor/bin/ece-tools cloud:config:update '{"stage":{"build":{"SCD_THREADS":3, "SCD_COMPRESSION_TIMEOUT":1000}}}'
```

> The configuration file will contain updated values

Reviewer: ```suggestion
The configuration file will contain updated values :
```

```yaml
stage:
  build:
    SCD_THREADS: 3
    SCD_COMPRESSION_TIMEOUT: 1000
  deploy:
    CLEAN_STATIC_FILES: false

> ```

Reviewer 2: Providing the following suggestion inline in this code comment because the Suggestion feature doesn't work properly when the suggestion includes closing code block (``` )

Consider replacing lines 80 - 113 with the following text:

Both commands require a single argument, a JSON-formatted array that specifies a value for at least one build, deploy, or post-deploy variable.  For example, the following command sets values for the `SCD_THREADS` and `CLEAN_STATIC_FILES` variables.

```bash
php vendor/bin/ece-tools cloud:config:create '{"stage":{"build":{"SCD_THREADS":5}, "deploy":{"CLEAN_STATIC_FILES":false}}}'
```
This command creates a new `.magento.env.yaml` file with the following settings:

```yaml
stage:
  build:
    SCD_THREADS: 5
  deploy:
    CLEAN_STATIC_FILES: false
```
You can use the `cloud:config:update` command to update the new file. For example, the following command changes the `SCD_THREADS` value and adds the `SCD_COMPRESSION_TIMEOUT` configuration:

```bash
php vendor/bin/ece-tools cloud:config:update '{"stage":{"build":{"SCD_THREADS":3, "SCD_COMPRESSION_TIMEOUT":1000}}}'
```

The updated file contains the following configuration:

```yaml
stage:
  build:
    SCD_THREADS: 3
    SCD_COMPRESSION_TIMEOUT: 1000
  deploy:
    CLEAN_STATIC_FILES: false
```

Author: Updated
